File: frontend/src/shared/models/sub-flow/sub-flow-dto.ts
import type { FlowSearchAreaDto } from "@/shared/models/flow-search-area/flow-search-area-dto";
import type { FlowStepDto } from "@/shared/models/flow-step/flow-step-dto";

export class SubFlowDto {
  id: number = -1;
  name: string = "";
  orderNumber: number = -1;

  flowSteps: FlowStepDto[] = [];
  flowSearchAreas: FlowSearchAreaDto[] = [];
}

File: backend/Core/Models/Dtos/Database/SubFlow/SubFlowDto.cs
using System.Collections.ObjectModel;

namespace Core.Models.Dtos
{
    public class SubFlowDto
    {
        public int Id { get; set; }

        public string Name { get; set; } = string.Empty;
        public int OrderNumber { get; set; }

        public virtual IEnumerable<FlowStepDto> FlowSteps { get; set; } = new Collection<FlowStepDto>();
        public virtual IEnumerable<FlowSearchAreaDto> FlowSearchAreas { get; set; } = new Collection<FlowSearchAreaDto>();
    }
}
